code/game/machinery/doppler_array.dm

```diff
@@ -194,44 +190,65 @@
 
 /obj/machinery/doppler_array/research
 	name = "tachyon-doppler research array"
-	desc = "A specialized tachyon-doppler bomb detection array that uses the results of the highest yield of explosions for research."
+	desc = "A specialized tachyon-doppler bomb detection array that uses complex on-board software to record data for experiments."
+	circuit = /obj/item/circuitboard/machine/doppler_array
 	var/datum/techweb/linked_techweb
 
+/obj/machinery/doppler_array/research/Initialize()
+	..()
+	linked_techweb = SSresearch.science_tech
+	return INITIALIZE_HINT_LATELOAD
+
+// Late initialize to allow the server machinery to initialize first
+/obj/machinery/doppler_array/research/LateInitialize()
+	. = ..()
+	AddComponent(/datum/component/experiment_handler, \
+		allowed_experiments = list(/datum/experiment/explosion), \
+		config_mode = EXPERIMENT_CONFIG_UI, \
+		config_flags = EXPERIMENT_CONFIG_ALWAYS_ACTIVE)
+
+/obj/machinery/doppler_array/research/attackby(obj/item/I, mob/user, params)
+	if (default_deconstruction_screwdriver(user, "tdoppler", "tdoppler", I) \
+		|| default_deconstruction_crowbar(I))
```

Comment:
```suggestion
/obj/machinery/doppler_array/research/attackby(obj/item/I, mob/user, params)
	if (default_deconstruction_screwdriver(user, "tdoppler", "tdoppler", I)
		|| default_deconstruction_crowbar(I)
	)
```

I'm pretty sure the backslash is not necessary (only needed for macros), and the parenthese on its own line makes it easier to scan and realize where the if statement ends.